src/pages/_app.tsx

```diff
@@ -55,18 +59,26 @@ function MyApp({
         `,
         }}
       />
-      <QueryClientProvider client={queryClient}>
-        <Hydrate state={pageProps.dehydratedState}>
-          <SessionProvider session={pageProps.session}>
-            <HeadMeta />
-            <ThemeProvider theme={theme}>
-              <GlobalStyle />
-              <HeaderBar />
-              <Component {...pageProps} />
-            </ThemeProvider>
-          </SessionProvider>
-        </Hydrate>
-      </QueryClientProvider>
+      <RecoilRoot>
+        <QueryClientProvider client={queryClient}>
+          <Hydrate state={pageProps.dehydratedState}>
+            <SessionProvider session={pageProps.session}>
+              <HeadMeta />
+              <ThemeProvider theme={theme}>
+                <GlobalStyle />
+                <HeaderBar />
+                {Component?.auth ? (
+                  <Auth>
+                    <Component {...pageProps} />
+                  </Auth>
+                ) : (
+                  <Component {...pageProps} />
+                )}
```

Comment:
auth 변수가 로그인이 필요한 페이지를 알려주는 flag라면 이게 더 맞을 것 같네요

```suggestion
                {Component?.needAuth ? (
                  <Auth>
                    <Component {...pageProps} />
                  </Auth>
                ) : (
                  <Component {...pageProps} />
                )}
```

Reply:
변경하겠습니다.!